src/main/java/data/exceptions/StorageOperationException.java

```diff
@@ -0,0 +1,10 @@
+package data.exceptions;
+
+public class StorageOperationException extends Exception {
+    public StorageOperationException() {
```

Comment:
(not just for this class) There are many empty constructors that you can consider leaving out?